internal/encoding/hex_test.go

```diff
@@ -103,3 +119,26 @@ func Test_DecodeZeroX(t *testing.T) {
 		})
 	}
 }
+
+func Test_EncodeHex(t *testing.T) {
+	for i, test := range encDecTests {
+		s := EncodeHex(test.dec)
+		if s != test.enc {
+			t.Errorf("#%d got:%s want:%s", i, s, test.enc)
+		}
+	}
+
+}
```

Comment:
please put the table for the table tests inside the test method. Look at this method as an example, you will need to change the test cases, method and text.
```
assert := assert.New(t)
type args struct {
		in string
	}
	tests := []struct {
		name    string
		args    args
		want    []byte
		wantErr bool
	}{
		{
			"success",
			args{
				"0x5602ea95540bee46d03ba335eed6f49d117eab95c8ab8b71bae2cdd1e564a761",
			},
			testutil.MustHexDecodeString("5602ea95540bee46d03ba335eed6f49d117eab95c8ab8b71bae2cdd1e564a761"),
			false,
		},
		{
			"err-missing-prefix",
			args{
				"5602ea95540bee46d03ba335eed6f49d117eab95c8ab8b71bae2cdd1e564a761",
			},
			nil,
			true,
		},
		{
			"err-empty",
			args{
				"",
			},
			nil,
			true,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := DecodeZeroX(tt.args.in)
			if (err != nil) != tt.wantErr {
				t.Errorf("decodeZeroX() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if !assert.DeepEqual(tt.want, got) {
				t.Errorf("decodeZeroX() = %v, want %v", got, tt.want)
			}
		})
	}
```